python/GangaLHCb/Lib/Applications/GaudiExec.py

```diff
@@ -204,11 +204,11 @@ def prepare(self, force=False):
         # file is unspecified, has a space or is a relative path
         self.configure(self)
         logger.info('Preparing %s application.' % getName(self))
-        self.is_prepared = ShareDir()
-        logger.info('Created shared directory: %s' % (self.is_prepared.name))
 
         this_build_target = self.buildGangaTarget()
 
+        self.is_prepared = ShareDir()
+        logger.info('Created shared directory: %s' % (self.is_prepared.name))
```

Comment:
I thought the build target depended on the target dir existing (else you might end up with one being inconsistent, I forget the logic but just moving this seems potentially bad from the watching what file is created when) 
Surely a try around the build target which removes the prepared dir (and from the prep repo) once any exception is thrown is safer?